src/ccl_halomod.c

```diff
@@ -0,0 +1,300 @@
+#include "ccl_utils.h"
+#include "math.h"
+#include "stdio.h"
+#include "stdlib.h"
+#include "gsl/gsl_integration.h"
+#include "gsl/gsl_interp.h"
+#include "gsl/gsl_spline.h"
+#include "gsl/gsl_sf_expint.h"
+#include "ccl_placeholder.h"
+#include "ccl_background.h"
+#include "ccl_power.h"
+#include "ccl_massfunc.h"
+#include "ccl_error.h"
+#include "class.h"
+#include "ccl_params.h"
+#include "ccl_emu17.h"
+#include "ccl_emu17_params.h"
+
+double delta_c(){
+  // Linear collapse threshold for haloes
+  return 1.686;
+}
+
+// TODO: referred to as odelta in ccl_massfunc, as a free parameter.
+double Delta_v() {
+  // Halo mean density
+  return 200.;
+}
+
+// TODO: Check growth factor is normalised g(a=1)=1
+double k_star(ccl_cosmology *cosmo, double a, int * status){
+  // HMcode parameter: k*  
+  double sigmaVz = ccl_sigmaV(cosmo, 0., status)*ccl_growth_factor(cosmo, a, status);
+  return 0.584/sigmaVz;
+}
+
+// TODO: Check growth factor is normalised g(a=1)=1
+double f_damp(ccl_cosmology *cosmo, double a, int * status){
+  // HMcode parameter: f  
+  double sigma8z = ccl_sigma8(cosmo, status)*ccl_growth_factor(cosmo, a, status);
+  return 0.188*pow(sigma8z, 4.29);
+}
+
+//TODO: Actually code this up
```

Comment:
This TODO should probably be included for the PR, no?
Or is this only for HMCode, not the generic halo model?